packages/office-addin-usage-data/src/usageData.ts

```diff
@@ -57,20 +53,31 @@ export class OfficeAddinUsageData {
   private usageDataClient = appInsights.defaultClient;
   private eventsSent: number = 0;
   private exceptionsSent: number = 0;
-  private options: IUsageDataOptions = {
-    groupName: defaults.groupName,
-    projectName: "ThisShouldBeOverwrittenInTheConstructor",
-    instrumentationKey: defaults.instrumentationKeyForOfficeAddinCLITools,
-    promptQuestion: `Office Add-in CLI tools collect anonymized usage data which is sent to Microsoft to help improve our product. Please read our privacy statement and usage data details at https://aka.ms/OfficeAddInCLIPrivacy.`,
-    raisePrompt: true,
-    usageDataLevel: UsageDataLevel.off,
-    method: UsageDataReportingMethod.applicationInsights,
-    isForTesting: false
-  }
+  private options: IUsageDataOptions;
+  private defaultData = {
+    Platform: process.platform,
+    NodeVersion: process.version
+  };
 
   constructor(usageDataOptions: IUsageDataOptions) {
     try {
-      this.overwriteProps(this.options, usageDataOptions);
+      this.options = {
+        groupName: defaults.groupName,
+        promptQuestion: "",
+        raisePrompt: true,
+        usageDataLevel: UsageDataLevel.off,
+        method: UsageDataReportingMethod.applicationInsights,
+        isForTesting: false,
+        ...usageDataOptions
+      };
+      
+      if (this.options.instrumentationKey === undefined) {
+        throw new Error("Instrumentation Key not defined - cannot create usage data object");
+      }
+
+      if (this.options.groupName === undefined) {
```

Comment:
effectively, this condition would never occur if we instantiate this.options.groupName to defaults.groupName above

Reply:
It's a safeguard in case a developer consuming our API passes in a `usageDataOptions` object with `groupName: undefined` to the constructor